src/silo/database.cpp

```diff
@@ -459,49 +533,53 @@ void Database::preprocessing(
 
    SPDLOG_INFO("preprocessing - building database");
 
-   build(preprocessing_config, partition_descriptor);
+   database.build(preprocessing_config, partition_descriptor, reference_genomes);
+
+   database.saveDatabaseState("output/serialized_state/", partition_descriptor, reference_genomes);
```

Comment:
I think I would like it better, if that function call was not in this method.
* It can be tested individually: doing preprocessing without writing it to the disc.
* Whoever call this method, can freely decide what to do with the resulting database, e.g. one can still call `saveDatabaseState`.

Reply:
Ah, here the problem is, for saving the database, one needs some datastructures which we want to discard once we finished the preprocessing to minimize coupling between preprocessing part and storage/querying part of the database

Reply:
I now removed all preprocessing structs from the database loading, which makes it more flexible anyways! (and reduced some coupling during the building of the database as well)